packages/ts-transform/src/utils/template-literal-to-css.tsx

```diff
@@ -41,59 +42,71 @@ export const templateLiteralToCss = (
         throw createNodeError('variable was not initialized', value);
       }
 
-      const variableName = cssVariableHash(value);
-
-      if (ts.isObjectLiteralExpression(value.initializer)) {
-        // We found an object expression e.g. const objVar = {}; css`${objVar}`
-        const result = objectLiteralToCssString(value.initializer, collectedDeclarations, context);
-        css += `${result.css}${span.literal.text.replace(/^;/, '')}`;
-        cssVariables = cssVariables.concat(result.cssVariables);
-      } else if (ts.isStringLiteral(value.initializer) || ts.isNumericLiteral(value.initializer)) {
-        const before = cssBeforeInterpolation(css);
-        // We an an inline arrow function - e.g. css`${props => props.color}`
-        const after = cssAfterInterpolation(span.literal.text);
-
-        css = before.css;
-        let cssVariableExpression: ts.Expression = span.expression;
-
-        if (after.variableSuffix && before.variablePrefix) {
-          cssVariableExpression = joinThreeExpressions(
-            ts.createStringLiteral(before.variablePrefix),
-            span.expression,
-            ts.createStringLiteral(after.variableSuffix)
-          );
-        } else if (after.variableSuffix) {
-          cssVariableExpression = joinToBinaryExpression(
-            span.expression,
-            ts.createStringLiteral(after.variableSuffix)
-          );
-        } else if (before.variablePrefix) {
-          cssVariableExpression = joinToBinaryExpression(
-            ts.createStringLiteral(before.variablePrefix),
-            span.expression
+      if (isConst(value)) {
```

Comment:
we might want to move this into the isStringLiteral/isNumbericLiteral if statement - else this might be inlining `object` and `function` stuff?

Reply:
got it!

Reply:
This reduced the large diff too, which was coming cuz of all the nesting of if-else!

Reply:
if you turn off whitespace the diff becomes way better

![image](https://user-images.githubusercontent.com/6801309/80657805-fb738b80-8ac7-11ea-839f-b4c3396e8484.png)